### DISCUSSION TITLE
Record audio and save in attachment field

### DISCUSSION THREAD

**Discussion by gabrielevaudano** [Budibase/budibase#6529] [Help]
Dear all,
My business application needs to record audio from the user (I was thinking of using MediaEncoder API available for almost all browsers) and then save the result inside a Budibase table attachment column. 

I tried to embed some HTML & JS code (using Embed element) but it doesn't work and neither it permits to record audio, while the same script works fine if it runs outside budibase.

Anyone knows a script to use to record audio & save it inside a row?
Thank you for your attention, kindly.

- MediaEncoder API for reference: https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_Recording_API/Using_the_MediaStream_Recording_API

**Comment by melohagan** [comment]
Hi @gabrielevaudano 

I think the embed component is not allowed to access the `window` object, which could be why the above doesn't work as it's using the navigator.

**Reply by gabrielevaudano** [reply]
Thank you @melohagan for your answer.
Could I auto-save audio in a form in another way inside an attachment column?
